models/status.go

```
@@ -231,7 +234,10 @@ func newCommitStatus(sess *xorm.Session, opts NewCommitStatusOptions) error {
 
 	// Insert new CommitStatus
 	if _, err = sess.Insert(opts.CommitStatus); err != nil {
-		sess.Rollback()
+		err = sess.Rollback()
```

The first error is lost.